### ISSUE TITLE
Regression? publication info dialog, long description does not have the expand/collapse button anymore

### ISSUE THREAD

**Issue by danielweck** [readium/readium-desktop#689]
![ThoriumOPDSDescriptionLongExpand](https://user-images.githubusercontent.com/625201/64435621-b757f480-d0ba-11e9-994b-27adb15e288c.png)

**Comment by danielweck** [comment]
https://github.com/readium/readium-desktop/blob/bfda8ee8de4a0c2b6005bcfc9ac5d4415c8bd54f/src/renderer/components/publication/publicationInfos/PublicationInfo.tsx#L200-L207

https://github.com/readium/readium-desktop/blob/bfda8ee8de4a0c2b6005bcfc9ac5d4415c8bd54f/src/renderer/components/publication/publicationInfos/PublicationInfo.tsx#L142-L164

**Comment by danielweck** [comment]
Culprit: `componentDidMount()` calls `needSeeMoreButton()` too early, CSS `offsetHeight` is not ready (the computation needs to occur post-rendering):

https://github.com/readium/readium-desktop/blob/bfda8ee8de4a0c2b6005bcfc9ac5d4415c8bd54f/src/renderer/components/publication/publicationInfos/PublicationInfo.tsx#L64-L69


I'm not sure why it worked before, maybe because of `componentDidUpdate()` which also calls `needSeeMoreButton()`?

https://github.com/readium/readium-desktop/blob/bfda8ee8de4a0c2b6005bcfc9ac5d4415c8bd54f/src/renderer/components/publication/publicationInfos/PublicationInfo.tsx#L71-L74

**Comment by danielweck** [comment]
`setTimeout()` fixes this bug (ensures CSS render pass is complete before comparing `offsetHeight`):

```
    public componentDidMount() {
        if (this.props.publicationIdentifier) {
            this.props.getPublicationFromId();
        }
        setTimeout(() => {
            this.needSeeMoreButton();
        }, 500);
    }
```

**Comment by danielweck** [comment]
@Scarsniik if you have a few minutes left before you go offline, would you mind looking into this? Is there a stage in the React rendering flow that we could use to ensure `needSeeMoreButton()` is called only once the HTML/CSS is rendered and ready?

**Comment by danielweck** [comment]
I don't see any obvious "blame" commit that would explain the regression:
https://github.com/readium/readium-desktop/blame/develop/src/renderer/components/publication/publicationInfos/PublicationInfo.tsx

Original PR:
https://github.com/readium/readium-desktop/pull/441

**Comment by Scarsniik** [comment]
Do you have the same problem in the library view or only on an opds ? I don't have any publication with long description to test

**Comment by Scarsniik** [comment]
I don't know how to use the function when all is ready.

**Comment by danielweck** [comment]
I tested in bookshelf (not OPDS) with long description. It works because of `componentDidUpdate()` (state propagation which the OPDS view not have). See my comments above.

So, maybe a timeout is fine for now, with a code comment to indicate that this is a hack until we find a reliable React event to trigger CSS computations.

**Comment by Scarsniik** [comment]
Ok, I'll do it like that for the moment

**Comment by danielweck** [comment]
Wait, the fix is simple!
"ref callback"

https://github.com/readium/readium-desktop/blob/bfda8ee8de4a0c2b6005bcfc9ac5d4415c8bd54f/src/renderer/components/publication/publicationInfos/PublicationInfo.tsx#L145

**Comment by Scarsniik** [comment]
I don't understand what you want to do with it

**Comment by danielweck** [comment]
Ah, maybe not, because the CSS layout pass is not complete yet at that stage (it's like the difference between the DOM "ready" event `DOMContentLoaded` vs. the window `load` event)

**Comment by danielweck** [comment]
Yep, looking at StackOverflow (etc.), `window.addEventListener('load')` and `setTimeout(() => {}, 1)` (aka "defer" function for next tick) are workable solutions :(

**Comment by danielweck** [comment]
https://github.com/facebook/react/issues/13108#issuecomment-491111789

**Comment by Scarsniik** [comment]
Ok, I'll use the timeout

**Comment by danielweck** [comment]
New issue: https://github.com/readium/readium-desktop/issues/759
